src/pybind/mgr/cephadm/services/iscsi.py

```diff
@@ -0,0 +1,105 @@
+            })
+
+        if spec.ssl_key:
+            if isinstance(spec.ssl_key, list):
+                key_data = '\n'.join(spec.ssl_key)
+            else:
+                key_data = spec.ssl_key
+            ret, out, err = self.mgr.mon_command({
+                'prefix': 'config-key set',
+                'key': f'iscsi/{utils.name_to_config_section("iscsi")}.{igw_id}/iscsi-gateway.key',
+                'val': key_data,
+            })
+
+        api_secure = 'false' if spec.api_secure is None else spec.api_secure
+        igw_conf = f"""
+        # generated by cephadm
+        [config]
+        cluster_client_name = {utils.name_to_config_section('iscsi')}.{igw_id}
+        pool = {spec.pool}
+        trusted_ip_list = {spec.trusted_ip_list or ''}
+        minimum_gateways = 1
+        api_port = {spec.api_port or ''}
+        api_user = {spec.api_user or ''}
+        api_password = {spec.api_password or ''}
+        api_secure = {api_secure}
+        """
+        extra_config = {'iscsi-gateway.cfg': igw_conf}
+        return self.mgr._create_daemon('iscsi', igw_id, host, keyring=keyring,
+                                   extra_config=extra_config)
+
+    def daemon_check_post(self, daemon_descrs: List[DaemonDescription]):
+        try:
+            _, out, _ = self.mgr.check_mon_command({
+                'prefix': 'dashboard iscsi-gateway-list'
+            })
+        except MonCommandFailed as e:
+                logger.warning('Failed to get existing iSCSI gateways from the Dashboard: %s', e)
+                return
+
+        gateways = json.loads(out)['gateways']
+        for dd in daemon_descrs:
+            spec = cast(IscsiServiceSpec,
+                        self.mgr.spec_store.specs.get(dd.service_name(), None))
+            if not spec:
+                logger.warn('No ServiceSpec found for %s' % dd)
+                continue
+            if not all([spec.api_user, spec.api_password]):
+                reason = 'api_user or api_password is not specified in ServiceSpec'
+                logger.warn('Unable to add iSCSI gateway to the Dashboard for %s: %s', dd, reason)
+                continue
+            host = self.mgr.inventory.get_addr(dd.hostname)
+            service_url = 'http://{}:{}@{}:{}'.format(
+                spec.api_user, spec.api_password, host, spec.api_port or '5000')
+            gw = gateways.get(dd.hostname)
+            if not gw or gw['service_url'] != service_url:
+                try:
+                    logger.info('Adding iSCSI gateway %s to Dashboard', service_url)
+                    ret, out, err = self.mgr.check_mon_command({
+                        'prefix': 'dashboard iscsi-gateway-add',
+                        'service_url': service_url,
+                        'name': dd.hostname
+                    })
+                except MonCommandFailed as e:
+                    logger.warn('Failed to add iSCSI gateway %s to the Dashboard: %s', service_url, e)
```

Comment:
```suggestion
                    logger.warning('Failed to add iSCSI gateway %s to the Dashboard: %s', service_url, e)
```

Reply:
Thanks, my brain didn't work quite well when writing and read the wrong deprecated one. Will fix it!